### DISCUSSION TITLE
add more SI units to volume and volume_storage device_class(es)

### DISCUSSION THREAD

**Discussion by bradkeifer** [home-assistant/architecture#1043] [Entity Models]
The volume and volume_storage device classes currently support the following SI units of measurement:

- L - litres
- mL - millilitres

I have a requirement for a sensor and also dashboard display of volumes in kilolitres (kL) and megalitres (ML). I make use of the modbus integration to obtain my sensor data, and I was alerted to the lack of support for the kL unit of measurement when reviewing the home-assistant.log file at startup, where the following warning message was given:
```
2024-02-21 10:32:38.029 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.freddy_actual_volume_r (<class 'homeassistant.components.template.sensor.SensorTemplate'>) is using native unit of measurement 'kl' which is not a valid unit for the device class ('volume') it is using; expected one of ['gal', 'fl. oz.', 'L', 'mL', 'm³', 'ft³', 'CCF']; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+template%22
```
The configuration I have for this sensor is:
```
# Flow Metering
    - name: freddy_actual_volume_r
      unit_of_measurement: kl
      device_class: volume
      state: >-
        {{ (states('sensor.dz_18_1920')|int|bitwise_and(4294901760) % 65535) + 
            (states('sensor.dz_18_1920')|int|bitwise_and(65535) / 1000)
        }}
```

I would like to see kilolitres and megalitres added to the set of units_of_measurement for the volume and volume_storage device_classes please.

**Comment by MartinHjelmare** [comment]
A cubic meter is the same as a thousand liters. Why don't you use cubic meter as unit?

What home application do you have that measures in million litres?

**Reply by parautenbach** [reply]
Because, for example, I have a water meter, and I can specify the flow rate as L/min, but I don't want to track the total as thousands of L (rather kL), but now I must use mˆ3. Large tanks and demostic water consumption isn't typially notated in L (not where I live, anyway).

**Reply by elupus** [reply]
Im also missing the knx native L/h

**Reply by MartinHjelmare** [reply]
My question was:

> What home application do you have that measures in million litres?

**Reply by MartinHjelmare** [reply]
If you're trying to answer this question: 

> Why don't you use cubic meter as unit?

I don't see any motivation why cubic meters isn't good enough.

**Reply by parautenbach** [reply]
And I answered this one:

> Why don't you use cubic meter as unit?

**Reply by bradkeifer** [reply]
> My question was:
> 
> 
> 
> > What home application do you have that measures in million litres?

On my home farm, which I understand is not a common case.

@MartinHjelmare are there architectural guidelines that apply here? If so, can you please point me to them.
